resolwe_bio/kb/filters.py

```diff
@@ -5,11 +5,114 @@
 =======
 
 """
+from functools import reduce
+
 import django_filters as filters
+from django.contrib.postgres.search import SearchQuery, SearchRank
+from django.db import models
+from django.db.models import F
 
 from resolwe.flow.filters import TEXT_LOOKUPS
 
-from .models import Mapping
+from .models import Feature, Mapping
+
+
+class FullTextFilter(filters.BaseCSVFilter):
+    """Filter for full-text search."""
+
+    def filter(self, queryset, value):
+        """Filter field by text-search vector.
+
+        Values, given in csv format are joined with the OR operator.
+
+        Filtering by more than 135 values raises recursion error, so we have
+        to do it in chunks.
+        """
+        if not value:
+            return queryset
+
+        value = list(map(lambda v: SearchQuery(v, config="simple"), value))
```

Comment:
Is not the list comprehension nicer is this situation?

`value = [SearchQuery(v, config="simple") for v in value]`